metadata/lib.rs

```diff
@@ -259,6 +241,120 @@ impl User {
     }
 }
 
+/// Builder for contract metadata
+#[derive(Default)]
+pub struct ContractBuilder {
+    name: Option<String>,
+    version: Option<Version>,
+    authors: Option<Vec<String>>,
```

Comment:
I think it is better to have just `Vec<String>` here since an empty vector is basically similar to no authors. The `authors` setter should check that the given `authors` parameter is non-empty though and that the already set `authors` is still empty.

Reply:
I have this as an option to make this pattern possible: https://github.com/paritytech/cargo-contract/blob/50b585b8da8856fe291d1936316729d64a99329b/metadata/lib.rs#L346